jina/docker/hubio.py

```diff
@@ -405,33 +410,47 @@ def build(self) -> Dict:
         return result
 
     def _test_build(self, image):
+        p_names = []
+        failed_levels = []
+
         # test uses at executor level
         if self.args.test_level >= BuildTestLevel.EXECUTOR:
-            with BaseExecutor.load_config(self.config_yaml_path):
-                pass
+            try:
+                with BaseExecutor.load_config(self.config_yaml_path):
+                    pass
+            except:
+                failed_levels.append(BuildTestLevel.EXECUTOR)
 
         # test uses at Pod level (no docker)
         if self.args.test_level >= BuildTestLevel.POD_NONDOCKER:
-            with Pod(set_pod_parser().parse_args(['--uses', self.config_yaml_path])):
-                pass
+            try:
+                with Pod(set_pod_parser().parse_args(['--uses', self.config_yaml_path])):
+                    pass
+            except:
+                failed_levels.append(BuildTestLevel.POD_NONDOCKER)
 
-        p_names = []
         # test uses at Pod level (with docker)
         if self.args.test_level >= BuildTestLevel.POD_DOCKER:
             p_name = random_name()
-            with Pod(set_pod_parser().parse_args(['--uses', image.tags[0], '--name', p_name] +
-                                                 ['--daemon'] if self.args.daemon else [])):
-                pass
-            p_names.append(p_name)
+            try:
+                with Pod(set_pod_parser().parse_args(['--uses', image.tags[0], '--name', p_name] +
+                                                    ['--daemon'] if self.args.daemon else [])):
+                    pass
+                p_names.append(p_name)
+            except:
+                failed_levels.append(BuildTestLevel.POD_DOCKER)
 
         # test uses at Flow level
         if self.args.test_level >= BuildTestLevel.FLOW:
             p_name = random_name()
-            with Flow().add(name=random_name(), uses=image.tags[0], daemon=self.args.daemon):
-                pass
-            p_names.append(p_name)
+            try:
+                with Flow().add(name=random_name(), uses=image.tags[0], daemon=self.args.daemon):
+                    pass
```

Comment:
FYI, `pass` won't run the flow. Is it what we expected?

Reply:
I think it is intended, it would be cool to be able to send some data to ensure it does not break in the `exec` method though. But it may be hard to provide a solution for every executor family, although we could provide some predefined set of input for tests. (Like text for nlpcrafters, docs with `matches` for rankers etc... etc...). It would be out of the scope of this PR however...